core/src/main/scala/akka/kafka/internal/ConsumerStage.scala

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.kafka.internal
+
+import java.util
+import java.util.concurrent.TimeoutException
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+import scala.concurrent.Promise
+import scala.concurrent.duration._
+import akka.Done
+import akka.stream._
+import akka.stream.scaladsl._
+import akka.stream.stage._
+import akka.kafka.scaladsl.Consumer
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.kafka.clients.consumer._
+import org.apache.kafka.common.TopicPartition
+import akka.kafka.scaladsl.Consumer.CommittableOffsetBatch
+
+/**
+ * INTERNAL API
+ */
+private[kafka] object ConsumerStage {
+
+  final case class CommittableOffsetImpl(offset: Consumer.Offset)(val stage: Committer) extends Consumer.CommittableOffset {
+    override def commit(): Future[Done] =
+      if (stage == null) Future.failed(new IllegalStateException(s"commit not supported, $offset"))
+      else stage.commit(offset)
+  }
+
+  trait Committer {
+    def commit(offset: Consumer.Offset): Future[Done]
+    def commit(offsets: Map[TopicPartition, OffsetAndMetadata]): Future[Done]
+  }
+
+  final case class CommittableOffsetBatchImpl(override val offsets: Map[TopicPartition, OffsetAndMetadata])(stage: Option[Committer])
+      extends CommittableOffsetBatch {
+
+    def updated(committableOffset: Consumer.CommittableOffset): CommittableOffsetBatch = {
+      val s = stage match {
+        case s @ Some(_) => s
```

Comment:
I'm not sure I get what you mean. It's ok to commit to any stage, also with a batch like this that was built up from many `CommittableOffsetImpl`. I'm just using the first stage reference.